pdr_backend/pdr_dashboard/test/test_callbacks_feeds.py

```diff
@@ -0,0 +1,181 @@
+import time
+
+from selenium.webdriver.common.by import By
+from pdr_backend.pdr_dashboard.test.resources import (
+    _navigate_to_feeds_page,
+    _assert_table_row_count,
+    start_server_and_wait,
+    _clear_filters,
+    _set_dropdown_and_verify_row_count,
+    _set_input_value_and_submit,
+    _remove_tags,
+    _set_searchbar_value,
+)
+
+
+def test_feeds_table(setup_app_with_etl_sample_data, dash_duo):
+    app = setup_app_with_etl_sample_data
+    start_server_and_wait(dash_duo, app)
+
+    _navigate_to_feeds_page(dash_duo)
+    dash_duo.wait_for_element("#feeds_page_table table")
+
+    table = dash_duo.find_element("#feeds_page_table table")
+
+    # Validate row and column count
+    rows = table.find_elements(By.XPATH, ".//tr")
+    assert len(rows) == 21
+
+    header = table.find_element(By.XPATH, ".//tr[1]")
+    columns = header.find_elements(By.XPATH, ".//th")
+    assert len(columns) == 12
+
+    # Validate headers
+    header_texts = [_remove_tags(c.text) for c in columns]
+    expected_headers = [
+        "",
+        "Addr",
+        "Base Token",
+        "Quote Token",
+        "Exchange",
+        "Time",
+        "Avg Accuracy",
+        "Avg Stake Per Epoch (Ocean)",
+        "Volume (Ocean)",
+        "Price (Ocean)",
+        "Sales",
+        "Sales Revenue (Ocean)",
+    ]
+    assert header_texts == expected_headers
+
+
+def test_feeds_page_metrics_row(setup_app_with_etl_sample_data, dash_duo):
+    app = setup_app_with_etl_sample_data
+    start_server_and_wait(dash_duo, app)
+
+    _navigate_to_feeds_page(dash_duo)
+    dash_duo.wait_for_element("#feeds_page_metrics_row")
+
+    metrics_row = dash_duo.find_element("#feeds_page_metrics_row")
+
+    # Validate metrics
+    # select first level divs
+    metrics = metrics_row.find_elements(By.XPATH, "./div")
+    assert len(metrics) == 5
+
+    metric_texts = [_remove_tags(m.text) for m in metrics]
+    expected_metrics = [
+        "Feeds",
+        "Accuracy",
+        "Volume",
+        "Sales",
+        "Revenue",
+    ]
+
+    for i, metric in enumerate(expected_metrics):
+        assert metric in metric_texts[i]
+
+
+def test_feeds_table_filters(setup_app_with_etl_sample_data, dash_duo):
+    app = setup_app_with_etl_sample_data
+    start_server_and_wait(dash_duo, app)
+
+    _navigate_to_feeds_page(dash_duo)
+    dash_duo.wait_for_element("#base_token")
+
+    # Test filtering with base token
+    _set_dropdown_and_verify_row_count(dash_duo, "#base_token", "ETH", 3)
+
+    # Test filtering with time
+    _set_dropdown_and_verify_row_count(dash_duo, "#time", "5m", 2)
+
+    _clear_filters(dash_duo)
+    _assert_table_row_count(dash_duo, "#feeds_page_table", 21)
```

Comment:
These asserts should be validating that you're getting the right columns and required data for all dashboards/work.

Asserting a simple row_count() is likely not appropriate coverage/validation

Reply:
I can agree that for the general table test but doing it for every filtering process is unnecessary

Reply:
ok I created a practical way to implement it and I developed this feature for all filter and search tests

Reply:
For example for filtering we could check that the table columns value respects the filter value: if filter by base token check that all base token column values are equal to filter value